aws/resource_aws_xray_sampling_rule.go

```diff
@@ -0,0 +1,221 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/xray"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsXraySamplingRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsXraySamplingRuleCreate,
+		Read:   resourceAwsXraySamplingRuleRead,
+		Update: resourceAwsXraySamplingRuleUpdate,
+		Delete: resourceAwsXraySamplingRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rule_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"resource_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"priority": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(1, 9999),
+			},
+			"fixed_rate": {
```

Comment:
@bflad Do you think this would require a custom validator as the validation helper does not support anything along the lined of `FloatBetween`?

Reply:
It appears there are two upstream PRs for adding `FloatBetween` to the `helper/validation` package upstream:

* https://github.com/hashicorp/terraform/pull/18614
* https://github.com/hashicorp/terraform/pull/20877

I cannot imagine there is a specific reason why one of these is not merged, so I will followup with those maintainers to make sure the new validation function gets in. We are likely pulling in a newer version of the upstream dependency later this week anyways as part of some Terraform 0.12 work, so this problem might resolve itself shortly. I would suggest omitting any changes in this regard for now.